tensorboard/plugins/debugger/README.md

```diff
@@ -0,0 +1,144 @@
+# The Debugger Dashboard
+
+The debugger dashboard offers a graphical user interface for the [TensorFlow debugger](https://www.tensorflow.org/versions/r1.1/programmers_guide/debugger). For instance, this dashboard
+
+* enables users to pause execution at specified ops or continue execution for some number of steps.
+* visualizes values of tensors.
+* associates tensors with specific lines in python code.
+
+![The Debugger Dashboard](images/debugger_intro.png)
+
+# Setup
+
+## Instrumenting Model Logic
+
+To use this dashboard, construct a `TensorBoardDebugWrapperSession`. Subsequently, that wrapper session will issue [gRPC](https://grpc.io/docs/guides/) messages to TensorBoard that containing data for debugging.
+
+The constructor accepts these parameters.
+
+1. the original `tf.Session` object.
+2. the `[[host]]:[[port]]` address to which to stream gRPC messages.
+
+Example logic:
+
+```python
+from tensorflow.python import debug as tf_debug
+sess = tf.Session()
+sess = tf_debug.TensorBoardDebugWrapperSession(sess, 'localhost:6064')
+sess.run(my_fetches)
+```
+
+## Start TensorBoard with the `debugger_port` flag.
+
+In addition to instrumenting the model, the user must pass the `debugger_port` flag to TensorBoard. This port number matches the one above. TensorBoard will then both receive gRPC messages from and issue gRPC messages to model logic.
+
+This example command demonstrates how to start TensorBoard:
+
+```
+tensorboard \
+    --logdir ~/project_foo/model_bar_logdir \
+    --port 6006 \
+    --debugger_port 6064 \
+;
+```
+
+## Navigating to TensorBoard
+
+Navigate to the debugger dashboard within TensorBoard based on which port it serves on (specified via the `port` flag). For instance, the URL might be `http://localhost:6006#debugger`.
+
+Initially, a dialog may indicate that the dashboard is waiting on a session run to begin. The dialog will hide once the latter happens.
+
+![Dialog on waiting for the first session run](images/debugger_dialog.png)
+
+# Selecting Ops
+
+To examine the output value of an op, ops in the graph must be first selected. Subsequently, the debugger dashboard will pause runs at those ops, enabling users to examine outputs of those ops.
+
+## The Node List
+
+Ops can be selected via the the node list on the top left:
```

Comment:
Can we try to be consistent in using the words "node" and "op" here? I know the terminology is confusing (even in tensorflow's own documentation), but I tend to refer to the elements the graph as "nodes" (hence the "Runtime Node List". I use the word "op" to refer to a class of "nodes", i.e., the class of all "MatMul" nodes, for example.